server/api/users.js

```diff
@@ -61,26 +63,49 @@ const apiLimiter = new RateLimit({
   },
 });
 
-router.post('/users/new', apiLimiter, multer.single('avatar'), async (req, res, next) => {
+function getBool(value = false) {
+  if (typeof value === 'string') {
+    return value !== 'false';
+  }
+  return value;
+}
+
+router.post('/users/new', apiLimiter, multer.single('avatarFile'), async (req, res, next) => {
   try {
     let payload;
     let firebaseUserId;
     const { platform } = req.body;
 
-    if (!platform) throw new ValidationError('INVALID_PLATFORM');
+    switch (platform) {
+      case 'wallet': {
+        const {
+          from,
+          payload: stringPayload,
+          sign,
+        } = req.body;
+        const isLogin = false;
+        payload = checkSignPayload(from, stringPayload, sign, isLogin);
+        break;
+      }
 
-    if (platform === 'wallet') {
-      const {
-        from,
-        payload: stringPayload,
-        sign,
-      } = req.body;
-      const isLogin = false;
-      payload = checkSignPayload(from, stringPayload, sign, isLogin);
-    } else {
-      const { firebaseIdToken } = req.body;
-      ({ uid: firebaseUserId } = await admin.auth().verifyIdToken(firebaseIdToken));
-      payload = req.body;
+      case 'email':
+      case 'google':
+      case 'twitter': {
+        const { firebaseIdToken } = req.body;
+        ({ uid: firebaseUserId } = await admin.auth().verifyIdToken(firebaseIdToken));
+        payload = req.body;
+        break;
+      }
+
+      case 'facebook': {
+        const { accessToken } = req.body;
+        await fetchFacebookUser(accessToken);
```

Comment:
should get verified info from `fetchFacebookUser()` and match/replace against untrusted payload from client